### DISCUSSION TITLE
Use custom pseudoselector in my theme

### DISCUSSION THREAD

**Discussion by carlkoudelka** [chakra-ui/chakra-ui#6128] [Q&A]
Hi,

I wanna to have an `Input` component with gray borders when `value` is empty, but `blue` when value is present.

I already have my own custom theme so my first idea was to do something like in my input variant. 

```typescript
  borderColor: props.value ? 'blue.600' : 'gray.600'
```
But the problem is that style is not refreshed when I actually change the value in my browser.

my second approach was something like this

```typescript
  borderColor: 'blue',
  '&[value=""]': { borderColor: 'gray' },
```

but that does not seems to work. I checked how `styled-system` works but I found no way to inject my own pseudoselectors to the system. I know I could try to make a PR and try to add new pseudoselector for `styled-system` but I hoped there is some more elegant way how to do it :).

**Comment by hugobqd** [comment]
You could simply use the **_borderColor_** props `<Input borderColor={props.value ? 'blue.600' : 'gray.600'} />`

I have to mention the **isInvalid** props too, because that's look like a king of validation mechanism :
```
<Input isInvalid errorBorderColor='red.300' />
```

More info https://chakra-ui.com/docs/components/form/input

**Reply by carlkoudelka** [reply]
Yes, I am aware of that, but I don't want to set it up for all inputs I have, neither I want to create an Input component which won't be part of chakra.

**Comment by hugobqd** [comment]
Oh! I understand now.

your CSS approche kinda work! See => https://codesandbox.io/s/relaxed-firefly-zsimpi?file=/src/index.js

But the inputs have to be controlled to update the 'value' html attribute.

**Reply by carlkoudelka** [reply]
I am confused why mine solution did not work since every input in my apps is controlled. But it looks promising.

What eludes me is which difference between controlled and uncontrolled input is responsible for this behaviour? This setting should be done on css level, right? What do I miss here?

**Reply by hugobqd** [reply]
For an uncontrolled element, the _value_ attribute in the rendered HTML is set on the initial render, and it will not change even when you type in the input. That's why the CSS could only target the initial state (provided by _defaultValue_ in the JSX).

For a controlled input, the _value_ attributes is updated in the dom.
